internal/provider/networks_wireless_ssids_resource.go

```diff
@@ -0,0 +1,1727 @@
+package provider
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/core-infra-svcs/terraform-provider-meraki/internal/provider/jsontypes"
+	"github.com/core-infra-svcs/terraform-provider-meraki/tools"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	openApiClient "github.com/meraki/dashboard-api-go/client"
+	"strings"
+)
+
+// TODO - DON'T FORGET TO DELETE ALL "TODO" COMMENTS!
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &NetworksWirelessSsidsResource{}
+	_ resource.ResourceWithConfigure   = &NetworksWirelessSsidsResource{}
+	_ resource.ResourceWithImportState = &NetworksWirelessSsidsResource{}
+)
+
+// TODO - This function needs to be added to the list of Data Sources in provider.go: func (p *ScaffoldingProvider) Resources
+// TODO - Otherwise the provider has no idea this resource exists.
+func NewNetworksWirelessSsidsResource() resource.Resource {
+	return &NetworksWirelessSsidsResource{}
+}
+
+// NetworksWirelessSsidsResource defines the resource implementation.
+type NetworksWirelessSsidsResource struct {
+	client *openApiClient.APIClient
+}
+
+type Dot11W struct {
```

Comment:
Nested structs in a resources data model should be named in a way that structs with common names are not overlapping in other resources. 

The best solution we have at this time is that all nested structs should be prefixed with a resource identifier. In this case: NetworksWirelessSsidsResourceModelDot11w

Reply:
Instead of prefixing all nested resource for model, can we just use the resource name for e.g Concentrator is nested object for Gre and Gre is at root, so instead of naming the Concetrator as ```NetworksWirelessSsidsResourceModelGreConcentrator``` can we go with ```NetworksWirelessSsidsResourceModelConcentrator```. This is avoid big struct names